### ISSUE TITLE
The function to get the embedding of dataset

### ISSUE THREAD

**Issue by Howardqlz** [brain-research/acai#11]
I want to use ACAI to pretrain, so i just want to use ACAI to generate the embedding of datasets(such as celeba), and i wonder what is the procedure?

And another question is the dimension of the embedding should be setup by which parameter?Or maybe the combination of which parameters?

And i didn't find the runs/classify.sh. Wish your feedback!

**Comment by craffel** [comment]
>  I want to use ACAI to pretrain, so i just want to use ACAI to generate the embedding of datasets(such as celeba), and i wonder what is the procedure?

cluster.py has some code for getting the latents for a dataset: https://github.com/brain-research/acai/blob/master/cluster.py#L84

> And another question is the dimension of the embedding should be setup by which parameter?Or maybe the combination of which parameters?

This is set by the `--latent` flag.

> And i didn't find the runs/classify.sh. Wish your feedback!

@david-berthelot any ideas on where this is?

**Comment by Howardqlz** [comment]
Yeah, i found that, thanks!
So if i want to generate a 256d embedding, it should be --latent=16 and --latent_width=4?
and for 64d， maybe --latent=16 and --latent_width=2?

**Comment by craffel** [comment]
To follow our experiments from the paper, keep --latent_width=4. For 256d, set --latent=16. For 64d, set --latent=4. etc.

**Comment by david-berthelot** [comment]
I think we removed `classify.sh` since we trained the classifier at the same time as the auto-encoder and reported results during the training process.

**Comment by Howardqlz** [comment]
Thanks for the help.  Now i use the "get_latents_and_labels" in cluster.py to generate the latent embedding of omniglot dataset, and i got a strange result as below:
    "Train/Test k-means objective = 238.5952 / 102.5819
     Train/Test accuracy 0.1437 / 0.142
     Rank 256 Inits 1000 Accuracy 14.21"
It seems that the result is not good. Since i run the mnist32 got the 0.96 for the acc.

Because the source code doesn't contain the preprocessing about omniglot dataset, i write it by myself and i am not sure my code is correct or not. 
The omniglot raw dataset consisted of two part, "backgroud" and "evaluation"(train and test). The "backgroud" part contain 19280 samples from 964 classes, each class has 20 samples. The "evaluation" part contain 13180 samples from 659 classes. Totally there are 1623 parts in omniglot dataset. And all these classes are from 50 datasets.
    For the result i write above, i segment the omniglot into two part,train and test. Both of the got the 1623 classes, the difference is the number of samples for each class. the ratio is 7:3.
And the i rewrite the lib/data.py like:
`_NCLASS = {
    'omniglot32': 50
}
`
`_DATASETS = {`
    `'omniglot32_train_once':`
        `functools.partial(`
            `input_fn_record,`
            `_parser_all,`
            `[os.path.join(DATA_DIR, 'omniglot-background.tfrecord')],`
            `shuffle=False,`
            `repeat=False,`
            `size=(105, 105, 1),`
            `resize=(32, 32)),`
    `'omniglot32_train':`
         `functools.partial(`
            `input_fn_record,`
            `_parser_all,`
            `[os.path.join(DATA_DIR, 'omniglot-background.tfrecord')],`
            `size=(105, 105, 1),`
            `resize=(32, 32)),`
    `'omniglot32_test':`
        `functools.partial(`
            `input_fn_record,`
            `_parser_all,`
            `[os.path.join(DATA_DIR, 'omniglot-evaluation.tfrecord')],`
            `shuffle=False,`
            `repeat=False,`
            `size=(105, 105, 1),`
            `resize=(32, 32))}`
And the ordered of the process is create_dataset.py、acai.py、cluster.py
So i don't know is omniglot can just got this low result or i have made some mistake
And i want to know when the metric of cluster.py is low, can i still use the latent embedding to do the classification or not.
Wish you could give me some advice, i would really appreciate it!

**Comment by david-berthelot** [comment]
I am glad you are able to reproduce the MNIST results, but I am not familiar with the new dataset you're trying to use. May be try to build some intuition by using a standard auto-encoder and looking at how reconstructions interpolate when only using a small latent size. 
This will help you sort where the problem might lie given that small latent sizes leads to better interpolation but worse reconstructions.

**Comment by Howardqlz** [comment]
Thanks for your reply and i will try to do what you say. 
Actually now i am confused about  one thing. If i have a dataset that it has 50 classes, and each class got 20 samples, that is totally 1000samples. And the i segment it into train(each class 16 samples) and test(each class 4 samples) set, then put into acai.  Can acai works well on this small dataset?

**Comment by david-berthelot** [comment]
I don't know, I am not familiar with the dataset. I have no intuition either way, there are many factors to consider. In the end the only way to know for sure is to try.
